lib/prelude.dx

```diff
@@ -42,45 +40,67 @@ def f_to_f32 (x : Float)   : Float32 = internal_cast _ x
 def i64_to_i (x : Int64)   : Int     = internal_cast _ x
 def i32_to_i (x : Int32)   : Int     = internal_cast _ x
 def w8_to_i  (x : Word8)   : Int     = internal_cast _ x
-def w8_to_n  (x : Word8)   : Nat     = internal_cast _ x
 def i_to_i64 (x : Int)     : Int64   = internal_cast _ x
 def i_to_i32 (x : Int)     : Int32   = internal_cast _ x
 def i_to_w8  (x : Int)     : Word8   = internal_cast _ x
 def i_to_w32 (x : Int)     : Word32  = internal_cast _ x
 def i_to_w64 (x : Int)     : Word64  = internal_cast _ x
 def w32_to_w64 (x : Word32): Word64  = internal_cast _ x
-def i_to_f (x:Int) : Float = internal_cast _ x
+def i_to_f (x:Int)   : Float = internal_cast _ x
+def f_to_i (x:Float) : Int   = internal_cast _ x
 def i64_to_raw_ptr (x:Int64 ) : RawPtr = internal_cast _ x
 def raw_ptr_to_i64 (x:RawPtr) : Int64  = internal_cast _ x
-def n_to_i64 (x : Nat) : Int64 = internal_cast _ x
-def n_to_i32 (x : Nat) : Int32 = internal_cast _ x
-def n_to_w32 (x : Nat) : Word32  = internal_cast _ x
-def n_to_w64 (x : Nat) : Word64  = internal_cast _ x
-def i64_to_n (x : Int64) : Nat = internal_cast _ x
-def i32_to_n (x : Int32) : Nat = internal_cast _ x
 
-def f_to_i (x:Float) : Int = internal_cast _ x
-def f_to_n (x:Float) : Nat = internal_cast _ x
-def n_to_f (x : Nat) : Float = internal_cast _ x
-def n_to_w8 (x : Nat) : Word8   = internal_cast _ x
+Nat = %Nat
+NatRep = Word32
+
+def nat_to_rep (x : Nat)    : NatRep = %projNewtype x
+def rep_to_nat (x : NatRep) : Nat    = %newtype Nat x
+
+def n_to_w8  (x : Nat) : Word8   = internal_cast _ $ nat_to_rep x
+def n_to_w32 (x : Nat) : Word32  = internal_cast _ $ nat_to_rep x
+def n_to_w64 (x : Nat) : Word64  = internal_cast _ $ nat_to_rep x
+def n_to_i32 (x : Nat) : Int32   = internal_cast _ $ nat_to_rep x
+def n_to_i64 (x : Nat) : Int64   = internal_cast _ $ nat_to_rep x
+def n_to_f32 (x : Nat) : Float32 = internal_cast _ $ nat_to_rep x
+def n_to_f64 (x : Nat) : Float64 = internal_cast _ $ nat_to_rep x
+def n_to_f   (x : Nat) : Float   = internal_cast _ $ nat_to_rep x
+
+def w8_to_n  (x : Word8)   : Nat = rep_to_nat $ internal_cast _ x
+def w32_to_n (x : Word32)  : Nat = rep_to_nat $ internal_cast _ x
+def w64_to_n (x : Word64)  : Nat = rep_to_nat $ internal_cast _ x
+def i32_to_n (x : Int32)   : Nat = rep_to_nat $ internal_cast _ x
+def i64_to_n (x : Int64)   : Nat = rep_to_nat $ internal_cast _ x
+def f32_to_n (x : Float32) : Nat = rep_to_nat $ internal_cast _ x
+def f64_to_n (x : Float64) : Nat = rep_to_nat $ internal_cast _ x
+def f_to_n   (x : Float)   : Nat = rep_to_nat $ internal_cast _ x
+
+interface FromUnsignedInteger a
```

Comment:
Is there a reason why we can't use `FromNat` anymore? It would be a lot less verbose and should work too?

Reply:
It was just to avoid having to add `nat_to_rep` in all the instances. But I think you're right. I'll change it back to `fromNat`.